<!-- a general launch file that should work on most systems -->

<launch>
    <node ns="camera" pkg="uvc_camera" type="camera_node" name="left_camera" output="screen">
		<param name="width" type="int" value="320" />
		<param name="height" type="int" value="240" />
		<param name="fps" type="int" value="30" />
		<param name="frame" type="string" value="left" />
		<param name="device" type="string" value="/dev/video0" />
		<param name="camera_info_url" type="string" value="file://$(find uvc_camera)/example-left.yaml" />
    </node>

	<node ns="camera" pkg="uvc_camera" type="camera_node" name="right_camera" output="screen">
		<param name="width" type="int" value="320" />
		<param name="height" type="int" value="240" />
		<param name="fps" type="int" value="30" />
		<param name="frame" type="string" value="right" />
		<param name="device" type="string" value="/dev/video1" />
		<param name="camera_info_url" type="string" value="file://$(find uvc_camera)/example-right.yaml" />
    </node>	
	
    <arg name="width" default="8" />
    <arg name="height" default="6" />
    <arg name="square" default="0.00615" />
    <arg name="left_topic" default="left_tracker" />
    <arg name="right_topic" default="right_tracker" />
	<arg name="left_camera_info" default="/left/camera_info" />
	<arg name="right_camera_info" default="/right/camera_info" />
	<arg name="left_camera_image" default="/left/image_color" />
	<arg name="right_camera_image" default="/right/image_color" />
 
    <node pkg="stereo_image_proc" type="stereo_image_proc" name="stereo_image_proc"/>

    <node pkg="tf" type="static_transform_publisher" name="camera_to_ground" args="0 0 0 0 0 0 ground camera 100" />
    <node pkg="tf" type="static_transform_publisher" name="left_to_right" args="0.5 0 0 1 0 0 left_camera right_camera 100" />

    <node name="left_chessboard_tracker" pkg="raven_2_vision" type="chessboard_tracker"
                args="--width $(arg width) --height $(arg height) --square $(arg square) --topic $(arg left_topic) --info $(arg left_camera_info) --image $(arg left_camera_image)"
                output="screen"/>

    <node name="right_chessboard_tracker" pkg="raven_2_vision" type="chessboard_tracker"
                args="--width $(arg width) --height $(arg height) --square $(arg square) --topic $(arg right_topic) --info $(arg right_camera_info) --image $(arg right_camera_image)"
                output="screen"/>

    <node name="pose_estimator" pkg="raven_pose_estimator" type="pose_estimator" output="screen">
		<remap from="left_poses" to="$(arg left_topic)" />
	 	<remap from="right_poses" to="$(arg right_topic)" />
		<remap from="marker_type" to="CHESS" />
    </node>



</launch>
